File: ProblemSet/463. 整数排序/java/463 · 整数排序 冒泡排序.java
public class Solution {
    /**
     * @param A: an integer array
     * @return: nothing
     */
    public void sortIntegers(int[] A) {
        for (int p = A.length - 1; p > 0; --p) {
            boolean exchange = false;
            for (int i = 0; i < p; ++i) {
                if (A[i] > A[i + 1]) {
                    int tmp = A[i];
                    A[i] = A[i + 1];
                    A[i + 1] = tmp;
                    exchange = true;
                }
            }
            if (!exchange) break;
        }
    }
}

File: ProblemSet/463. 整数排序/python/463 · 整数排序 冒泡排序.py
class Solution:
    """
    @param A: an integer array
    @return: nothing
    """
    def sortIntegers(self, A):
        for p in range(len(A) - 1, 0, -1):
            exchange = False
            for i in range(p):
                if A[i] > A[i + 1]:
                    A[i], A[i + 1] = A[i + 1], A[i]
                    exchange = True
            if not exchange:
                break